File: Chandra Kiran/20-09-2022/sortlist.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var sortList = function (head) {
    let temp = head;
    let i = 0;
    let arr = [];
    while (temp != null) {
        arr[i++] = temp.val;
        temp = temp.next;
    }
    arr.sort(key = (a, b) => a - b);
    temp = head;
    i = 0;
    while (temp != null) {
        temp.val = arr[i++];
        temp = temp.next;
    }
    return head;
};

File: Nithya/19-09-2002/sortlist.java
class Solution {
    public ListNode sortList(ListNode head) {
                   ListNode start=new ListNode();
           start.next=head;
           ListNode curr=head, prev=start;
           while(curr!=null)
           {
               if(curr.next!=null &&(curr.next.val<curr.val))
               {
                   while(prev.next!=null && (prev.next.val<curr.next.val))
                       prev=prev.next;
                   ListNode temp=prev.next;
                   prev.next=curr.next;
                   curr.next=curr.next.next;
                   prev.next.next=temp;
                   prev=start;
               }
               else
                   curr=curr.next;
           }
           return start.next;   
        
    }
}